File: java/src/array/ArrayStack.java
package array;

public class ArrayStack<E> {

    private E[] array;

    private int numElement;

    public ArrayStack() {
        array = (E[]) new Object[5];
    }

    public String toString() {
        StringBuffer sb = new StringBuffer();

        sb.append("[");
        for (int i = 0 ; i < numElement; i++) {
            sb.append(array[i]);
            sb.append(",");
        }
        sb.deleteCharAt(sb.length() - 1);
        sb.append("]");

        return sb.toString();
    }

    public void push(E input) {
        array[numElement++] = input;
    }

    public E top() {
        if (numElement == 0)
            return null;
        return array[numElement - 1];
    }

    public E pop() {
        if (numElement == 0)
            return null;

        E returnValue = array[--numElement];
        array[numElement] = null;

        return returnValue;
    }

    public boolean empty() {
        return numElement == 0;
    }
}

File: python/array/arrayStack.py
class ArrayStack:
    arr = []
    size = 0

    def __init__(self):
        self.arr = []
        self.size = 0

    def toString(self):
        result = []
        for i in range(self.size-1, -1, -1):
            result.append(str(self.arr[i]))
        return ','.join(result)

    def push(self, input):
        self.arr.append(input)
        self.size += 1

    def top(self):
        if self.size == 0:
            return None
        return self.arr[self.size - 1]

    def pop(self):
        if self.size == 0:
            return None
        self.size -= 1
        return self.arr[self.size]

    def empty(self):
        return self.size == 0

